plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/jobs/JobUtil.java

```diff
@@ -0,0 +1,45 @@
+package com.google.cloud.tools.eclipse.util.jobs;
+
+import org.eclipse.core.runtime.jobs.IJobChangeEvent;
+import org.eclipse.core.runtime.jobs.IJobChangeListener;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.jobs.JobChangeAdapter;
+import org.eclipse.e4.core.contexts.ContextInjectionFactory;
+import org.eclipse.e4.core.contexts.EclipseContextFactory;
+import org.eclipse.e4.core.contexts.IEclipseContext;
+import org.osgi.framework.FrameworkUtil;
+
+public class JobUtil {
```

Comment:
The main reason to add the method to create the job is that this way the context used for DI is disposed by the job listener. On the caller side it leads the developer to define the parameters while takes care of the context disposal which could be forgotten. But probably I've misused DI and indeed using directly constructors would be cleaner. I'll review it.